062-topic-replication-factor-change.md

```diff
@@ -0,0 +1,223 @@
+For that reason, we introduce the following environment variables, which will be set automatically by the Cluster Operator, or manually by the user in case of standalone deployment.
+
+| Name                                   | Type    | Default | Description                                           |
+|:---------------------------------------|---------|---------|-------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED         | Boolean | false   | Whether Cruise Control integration is enabled         |
+| STRIMZI_CRUISE_CONTROL_NAMESPACE       | String  | ""      | Namespace where Cruise Control is running             |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME        | String  | ""      | Cruise Control hostname                               |
+| STRIMZI_CRUISE_CONTROL_PORT            | Integer | 9090    | Cruise Control port                                   |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED     | Boolean | false   | Whether Cruise Control SSL encryption is enabled      |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED    | Boolean | false   | Whether Cruise Control Basic authorization is enabled |
+| STRIMZI_CRUISE_CONTROL_SECRET_NAME     | String  | ""      | Secret name containing Cruise Control truststore      |
+| STRIMZI_CRUISE_CONTROL_API_SECRET_NAME | String  | ""      | Secret name containing Cruise Control API credentials |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED    | Boolean | false   | Whether the target Kafka cluster has rack awareness   |
+
+The topic configuration endpoint is asynchronous, which means that it validates the task, but the execution will be queued.
+Inside Cruise Control, a valid topic configuration task is first create in an `Active` state, then moved to `InExecution`, and finally to `Completed` or `CompletedWithError`.
+Only a single task can be `InExecution` state at any given time, which includes both topic configuration and rebalance tasks.
+
+When Cruise Control integration is enabled, a `ReplicationFactorManager` is created to manage the communication with the REST API.
+The `ReplicationFactorManager` stores all ongoing replication factor changes in an in-memory cache, in order to periodically check their status.
+A single replication factor change can be ongoing at any given time for a specific `KafkaTopic` resource, following replication factor changes will remain pending, and periodically retried.
+At most `max.active.user.tasks` (default: 5) replication factor changes for different `KafkaTopic` resources can be ongoing at the same time.
+
+Cruise Control allows to scale down the replication factor under the `min.insync.replicas` value, and this can cause an availability issue for producers with `acks=all`.
+We won't implement any check for that, because the `Cluster Operator` ignores pathological topic configurations where RF < minISR, and these topics don't event show up as under replicated.
```

Comment:
It is fine for me to not enforce anything. But it would be nice to have a warning to let users know they have it misconfigured.

Reply:
Good idea. Thanks.